### PR TITLE
Fix shortcodes

### PR DESCRIPTION
**Pull request by chriscct7**
So the params of the shortcode can actually be used in a shortcode.

#67

### CODE REVIEW

**Comment by cklosowski** [comment]
@pippinsplugins, @chriscct7 is right, we need to support if someone runs the shortcode with a `user_id` attribute.

I'd rather see this done with `shortcode_atts`, but we might have issues with that, maybe we should only run it through the shortcode atts if it's an array? That way it's not reliant on a numerical index, in case someone decides to just add some odd parameters to it.
```
    $a = shortcode_atts( array(
        'foo' => 'something',
        'bar' => 'something else',
    ), $atts );
```

File: includes/short-codes.php [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
 
-	$user_id  = get_current_user_id();
 	$products = eddc_get_download_ids_of_user( $user_id );
 
-	if( ! is_user_logged_in() )
+	if( empty( $user_id ) ) {
+		$user_id = get_current_user_id();
+	}
+
+	if ( is_array( $user_id ) ){
```
Comment (pippinsplugins) [inline includes/short-codes.php]: Why is this supporting an array of user IDs?

Comment (chriscct7) [inline includes/short-codes.php]: Because shortcode atts come in as an array.

Comment (chriscct7) [inline includes/short-codes.php]: So like [shortcode user="1"] comes in as an array where the first index is
user id

Comment (chriscct7) [inline includes/short-codes.php]: That flattens it if it's an array

Comment (pippinsplugins) [inline includes/short-codes.php]: Then it's not being properly setup. See Chris C's comment on the issue.

**Comment by chriscct7** [comment]
You mean Chris k's. Yep we know. Just haven't updated it yet

**Comment by cklosowski** [comment]
I've addressed this in #72

**Pull request closed without merging**